scripts/research/lib/batch02_source_policy.py

```
@@ -0,0 +1,676 @@
+                    path,
+                    f"canonical Batch02 binding may not be shadowed by argument: "
+                    f"{node.arg}",
+                    getattr(node, "lineno", None),
+                )
+            )
+
+        if isinstance(node, ast.Attribute):
+            if (
+                isinstance(node.ctx, (ast.Store, ast.Del))
+                and node.attr in _PROTECTED_BINDINGS
+            ):
```

Ruff (0.16.2)

[warning] Use a single `if` statement instead of nested `if` statements

Combine `if` statements using `and`

(SIM102)